src/adera/machines/links.cpp

```diff
@@ -1,12 +1,63 @@
 #include "links.h"
 
+using namespace osp;
+
 using osp::link::MachTypeReg_t;
 using osp::link::MachTypeId;
 
 namespace adera
 {
 
-MachTypeId const gc_mtUserCtrl = osp::link::MachTypeReg_t::create();
-MachTypeId const gc_mtMagicRocket = osp::link::MachTypeReg_t::create();
+MachTypeId const gc_mtUserCtrl      = MachTypeReg_t::create();
+MachTypeId const gc_mtMagicRocket   = MachTypeReg_t::create();
+MachTypeId const gc_mtRcsDriver     = MachTypeReg_t::create();
+
+float thruster_influence(Vector3 const pos, Vector3 const dir, Vector3 const cmdLin, Vector3 const cmdAng) noexcept
```

Comment:
My reading of this function is that it's a constexpr candidate.

If not, consider https://stackoverflow.com/questions/29117836/attribute-const-vs-attribute-pure-in-gnu-c

`__attribute((const))`

since it only reads it's inputs and then returns a single output.

Reply:
thruster_influence requires a square root, so constexpr isn't possible (or just very difficult). `__attribute__((const))` seems really obscure, and I have doubts it would make any difference at this time; also noting LTO.